### PR TITLE
created a default value field for assets

### LINKED ISSUE #436: [Feature request]: Create a default value field for assets
### Contact Details

_No response_

### Is your feature request related to a problem? Please describe?

We want to show on the dashboard the total value of assets of the user database (or workspace database). 

Screenshot 2023-10-06 at 12 04 16

We cannot rely on a custom field for this. So we would like to do it with a standard field on shelf.

### Describe the solution you'd like

On asset create, have a value field - so our users fill it up and we can calculate it. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hunar1312**
resolves #436

### CODE REVIEW

File: app/database/migrations/20230428075403_add_qr_to_db/migration.sql
Comment (DonKoko) [inline app/database/migrations/20230428075403_add_qr_to_db/migration.sql]: We removed this for testing purposes when we were checking the issue you had. There should be no changes in this file. Please revert them

File: app/routes/_layout+/assets.$assetId.tsx
Comment (DonKoko) [inline app/routes/_layout+/assets.$assetId.tsx]: I dont think you need to do that. You can just adjust the query for asset by including organization and selecting only the currency field.
Something like this. 
```
include: {
  organization: {
    select: {
      currency: true
     },
  }
}
```

and then u can do `asset.organization.currency`

File: app/routes/_layout+/assets.new.tsx
Comment (DonKoko) [inline app/routes/_layout+/assets.new.tsx]: Not sure why we are passing the currency here. It comes from the loader, just get it inside the asset form itself.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Added some comments. Mostly small changes.

**Review by DonKoko — CHANGES_REQUESTED** [review]
So far so good, however we are missing some stuff.

- [x] displaying value on asset page
- [ ] Setting default currency

The first part you can do now already, but the second part is a bit tricky, as it has to be an option on workspace, but we are still working on the workspaces/organizations, so I think we have to wait for it to be merged before we can merge this as if we dont wait it will create db and code conflicts and its just not worth it. Lets resolve the first one and then mark this as blocked by #369

**Comment by hunar1312** [comment]
https://github.com/Shelf-nu/shelf.nu/pull/455/files#diff-6abae98ca8769c62228e2e3bed3a8388223caed03c9c1aab378bbb4b46e95d15 the first part is already done @DonKoko and we already discussed that we won't move froward till the organization's page is refactored. So yeah we gotta put hold on this one for now.

**Comment by DonKoko** [comment]
@hunar1312 yeah i also did some more testing and I realized we are really blocked by the organization refactoring so we have to wait before we can merge it.

**Comment by DonKoko** [comment]
@hunar1312 updated the PR by merging it with main. There are some changes that will need to be done and you will get some TS errors so better solve them first. Some of the types are moved to a different file as well as now everything is scoped with organizationId instead of userId.

**Comment by DonKoko** [comment]
@hunar1312 I have updated the branch with main. Here is what is still pending:

- [ ] code needs to be updated to use organizationId instead of userId for scoping of everything. I believe TS should already give you errors on those places where orgId is missing because its required.
- [ ] We have a view for editing workspaces. We should add a field where they can change workspace currency.

**Review by DonKoko — APPROVED** [review]

**Pull request merged**
